pilot/pkg/model/test/mockopenidserver.go

```diff
@@ -37,10 +37,16 @@ var (
 
 const (
 	// JwtPubKey1 is the response to 1st call for JWT public key returned by mock server.
-	JwtPubKey1 = "fakeKey1"
+	JwtPubKey1 = `{ "keys": [ { "kid": "fakeKey1_1", "alg": "RS256", "kty": "RSA", "n": "abc", "e": "def" },
+			{ "kid": "fakeKey1_2", "alg": "RS256", "kty": "RSA", "n": "123", "e": "456" } ] }`
+
+	// JwtPubKey1Reordered is the response to 1st call for JWT public key returned by mock server, but in a modified order of json elements.
+	JwtPubKey1Reordered = `{ "keys": [ { "alg": "RS256", "kid": "fakeKey1_2", "n": "123", "kty": "RSA", "e": "456" },
```

Comment:
Can we add a few more test cases:
- jwt that doesn't have "kid" or "keys"
- jwt that has less or more items in its map, for example, a key missing the "alg" item or with an extra "bla" item.

Reply:
Added tests for these specifically for the compareJWKSResponse method